awscli/clidocs.py

```diff
@@ -716,3 +730,62 @@ def _line_has_tag(self, line):
 
     def doc_subitems_start(self, help_command, **kwargs):
         pass
+
+
+class GlobalsRenderer:
+    """
+    Renderer used to generate global options for the
+    Synopsis and Global Options sections.
+    """
+
+    def __init__(self, driver):
+        self._driver = driver
+        self._global_options = None
+        self._global_synopsis = None
+
+    @property
+    def global_options(self):
+        if not self._global_options:
+            self._generate_global_options()
+        return self._global_options
+    
+    @property
+    def global_synopsis(self):
+        if not self._global_synopsis:
+            self._generate_global_synopsis()
+        return self._global_synopsis
+    
+    def _generate_global_options(self):
+        doc = ReSTDocument()
+        arg_table = self._driver._get_argument_table()
+        for arg in arg_table:
+            argument = arg_table.get(arg)
+            doc.writeln(f'``{argument.cli_name}`` ({argument.cli_type_name})')
+            doc.include_doc_string(argument.documentation)
+            if argument.choices:
+                doc.style.start_ul()
+                for choice in argument.choices:
+                    doc.style.li(choice)
+                doc.style.end_ul()
+        self._global_options = doc.getvalue()
+    
+    def _generate_global_synopsis(self):
+        doc = ReSTDocument()
+        arg_table = self._driver._get_argument_table()
+        for arg in arg_table:
+            argument = arg_table.get(arg)
+            option_str = '[%s]' % argument.cli_name
```

Comment:
It would be interesting to see if we can reuse this logic from other places where generate the non-global synopsis arguments. Mainly if we ever update that logic, we will want to make sure we are consistent across global and non-global options.

Reply:
I decided not to reuse logic for 2 reasons:
1. For global options, the logic was just copied over from `ProviderDocumentEventHandler`. The top-level global options had its own logic before this PR and that hasn't changed.
2. For global synopsis, the logic in `CLIDocumentEventHandler` is incompatible with global options. `elif argument.cli_name.startswith('--'):` would target all global options (they all start with `--`) but we definitely don't want all global options being processed under this conditional

Reply:
That's fine. I was more referring to point 2 here. Looking through the logic on `CLIDocumentEventHandler` I'm fine with having our own logic. Mainly it would take larger rework to reuse the logic in `CLIDocumentEventHandler`  to work with the top level arguments we generate and would prefer not to touch that logic for now.